signing/secp256k1/impl/src/main/java/tech/pegasys/signers/secp256k1/multikey/MultiKeyTransactionSignerProvider.java

```diff
@@ -39,72 +43,92 @@
 
-  public static MultiKeyTransactionSignerProvider create(final Path rootDir) {
+  public static MultiKeyTransactionSignerProvider create(
+      final Path rootDir, final FileSelector<PublicKey> configFileSelector) {
     final SigningMetadataTomlConfigLoader signingMetadataTomlConfigLoader =
         new SigningMetadataTomlConfigLoader(rootDir);
 
-    final AzureKeyVaultTransactionSignerFactory azureFactory =
-        new AzureKeyVaultTransactionSignerFactory(new AzureKeyVaultAuthenticator());
+    final AzureKeyVaultSignerFactory azureFactory =
+        new AzureKeyVaultSignerFactory(new AzureKeyVaultAuthenticator());
 
     final HashicorpSignerFactory hashicorpSignerFactory = new HashicorpSignerFactory(Vertx.vertx());
 
     return new MultiKeyTransactionSignerProvider(
-        signingMetadataTomlConfigLoader, azureFactory, hashicorpSignerFactory);
+        signingMetadataTomlConfigLoader, azureFactory, hashicorpSignerFactory, configFileSelector);
   }
 
   public MultiKeyTransactionSignerProvider(
       final SigningMetadataTomlConfigLoader signingMetadataTomlConfigLoader,
-      final AzureKeyVaultTransactionSignerFactory azureFactory,
-      final HashicorpSignerFactory hashicorpSignerFactory) {
+      final AzureKeyVaultSignerFactory azureFactory,
+      final HashicorpSignerFactory hashicorpSignerFactory,
+      final FileSelector<PublicKey> configFileSelector) {
     this.signingMetadataTomlConfigLoader = signingMetadataTomlConfigLoader;
     this.azureFactory = azureFactory;
     this.hashicorpSignerFactory = hashicorpSignerFactory;
+    this.configFileSelector = configFileSelector;
   }
 
   @Override
-  public Optional<TransactionSigner> getSigner(final String address) {
+  public Optional<TransactionSigner> getSigner(final PublicKey publicKey) {
     return signingMetadataTomlConfigLoader
-        .loadMetadataForAddress(address)
+        .loadMetadata(configFileSelector.getSpecificConfigFileFilter(publicKey))
         .map(metadataFile -> metadataFile.createSigner(this));
   }
 
   @Override
-  public Set<String> availableAddresses() {
-    return signingMetadataTomlConfigLoader.loadAvailableSigningMetadataTomlConfigs().stream()
-        .map(metadataFile -> metadataFile.createSigner(this))
+  public Set<PublicKey> availablePublicKeys() {
+    return signingMetadataTomlConfigLoader
+        .loadAvailableSigningMetadataTomlConfigs(configFileSelector.getCollectiveFilter()).stream()
+        .map(
+            metadataFile -> {
+              final TransactionSigner signer = metadataFile.createSigner(this);
+              try {
+                if ((signer != null)
+                    && configFileSelector
+                        .getSpecificConfigFileFilter(signer.getPublicKey())
+                        .accept(Path.of(metadataFile.getFilename()))) {
+                  return signer;
+                }
+                return null;
+              } catch (final IOException e) {
+                return null;
```

Comment:
The Filter forces there to be an "Throws IOException" - so it needs to be handled, but if it fails, the right answer for this function is to return null rather than letting the exception escape.